sklearn/metrics/_plot/tests/test_confusion_matrix_display.py

```diff
@@ -0,0 +1,328 @@
+from numpy.testing import (
+    assert_allclose,
+    assert_array_equal,
+)
+import numpy as np
+import pytest
+
+from sklearn.datasets import make_classification
+from sklearn.compose import make_column_transformer
+from sklearn.exceptions import NotFittedError
+from sklearn.linear_model import LogisticRegression
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.svm import SVC
+from sklearn.svm import SVR
+
+from sklearn.metrics import ConfusionMatrixDisplay
+from sklearn.metrics import confusion_matrix
+
+
+# TODO: Remove when https://github.com/numpy/numpy/issues/14397 is resolved
+pytestmark = pytest.mark.filterwarnings(
+    "ignore:In future, it will be an error for 'np.bool_':DeprecationWarning:"
+    "matplotlib.*"
+)
+
+
+@pytest.fixture(scope="module")
+def n_classes():
+    return 5
```

Comment:
I know it was there before but I'm not sure that declaring a fixture with module scope is necessary. Looks like this one could just be a global constant, and it's no big deal to just call `make_classification` in each test


Adding all of them as parameters to the `test_...` function adds noise IMHO. It also hurts readability because what a test does is now defined in various different places and we need to switch back and forth between the test definition and the top of the  file.

But maybe that's a matter of taste.